### ISSUE TITLE
App/UI crash when trying to download video

### ISSUE THREAD

**Issue by 5432123** [TeamNewPipe/NewPipe#8814]
### Checklist

- [X] I am able to reproduce the bug with the [latest version](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

0.23.2

### Steps to reproduce the bug

1. I go to a video
2. Press download
3. Press ok
4. The app crashes

### Expected behavior

I expect to be able to download a video.

### Actual behavior

The app crashes when I try to download a video.

## Exception
* __User Action:__ ui error
* __Request:__ ACRA report
* __Content Country:__ AU
* __Content Language:__ en-AU
* __App Language:__ en_AU
* __Service:__ none
* __Version:__ 0.23.2
* __OS:__ Linux samsung/heroltexx/herolte:8.0.0/R16NW/G930FXXU3ERL3:user/release-keys 8.0.0 - 26
<details><summary><b>Crash log </b></summary><p>

```
java.lang.NullPointerException: Attempt to get length of null array
	at android.os.Parcel.readException(Parcel.java:1973)
	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:183)
	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:135)
	at android.content.ContentProviderProxy.query(ContentProviderNative.java:418)
	at android.content.ContentResolver.query(ContentResolver.java:760)
	at android.content.ContentResolver.query(ContentResolver.java:710)
	at android.content.ContentResolver.query(ContentResolver.java:668)
	at org.schabi.newpipe.streams.io.StoredDirectoryHelper.findFileSAFHelper(StoredDirectoryHelper.java:302)
	at org.schabi.newpipe.streams.io.StoredDirectoryHelper.findFile(StoredDirectoryHelper.java:218)
	at org.schabi.newpipe.download.DownloadDialog.prepareSelectedDownload(DownloadDialog.java:763)
	at org.schabi.newpipe.download.DownloadDialog.lambda$initToolbar$1(DownloadDialog.java:331)
	at org.schabi.newpipe.download.DownloadDialog.$r8$lambda$QfMS-FoXWFjSoHiNV7JREecQBWg(Unknown Source:0)
	at org.schabi.newpipe.download.DownloadDialog$$ExternalSyntheticLambda6.onMenuItemClick(Unknown Source:2)
	at androidx.appcompat.widget.Toolbar$1.onMenuItemClick(Toolbar.java:208)
	at androidx.appcompat.widget.ActionMenuView$MenuBuilderCallback.onMenuItemSelected(ActionMenuView.java:780)
	at androidx.appcompat.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:834)
	at androidx.appcompat.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:158)
	at androidx.appcompat.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:985)
	at androidx.appcompat.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:975)
	at androidx.appcompat.widget.ActionMenuView.invokeItem(ActionMenuView.java:624)
	at androidx.appcompat.view.menu.ActionMenuItemView.onClick(ActionMenuItemView.java:151)
	at android.view.View.performClick(View.java:6897)
	at android.widget.TextView.performClick(TextView.java:12727)
	at android.view.View$PerformClick.run(View.java:26101)
	at android.os.Handler.handleCallback(Handler.java:789)
	at android.os.Handler.dispatchMessage(Handler.java:98)
	at android.os.Looper.loop(Looper.java:164)
	at android.app.ActivityThread.main(ActivityThread.java:6944)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)

```
</details>
<hr>

### Affected Android/Custom ROM version

Android 8.0

### Affected device model

Samsung Galaxy S7

**Comment by tadekdudek** [comment]
For me all is working perfectly. Using version 0.23.2 on LineageOS 18.1 (Android 11) on Redmi Note 7.

**Comment by dimy93** [comment]
Happens to me intermittently on one plus 8 pro with android 12

**Comment by mhmdanas** [comment]
Duplicate of #8760. The error was fixed in NewPipe 0.23.2.

**Comment by dimy93** [comment]
This as the original reporter has said happens on 0.23.2 and it is thus not fixed

**Comment by mhmdanas** [comment]
Yeah, my bad, I closed this issue prematurely.

**Comment by dimy93** [comment]
For me I cannot be 100% sure, but given that killing the app ( and its background processes) and relaunching it fixed it I would assume that was not the issue at hand. Also I had 0.23.1 for a while before the YouTube bug and I had downloaded videos there already. Also I want to add I first saw this issue on the debug app shared in the GitHub issue tracker that contained the 0.23.2 fixes and there for me the issue was happening always. On 0.23.2 itself it is only intermittent.
